src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -252,15 +252,27 @@ public async Task<HttpResponseMessage> SendAsync(CancellationToken cancellationT
                 {
                     case Http3ErrorCode.VersionFallback:
                         // The server is requesting us fall back to an older HTTP version.
-                        throw new HttpRequestException(SR.net_http_retry_on_older_version, ex, RequestRetryType.RetryOnLowerHttpVersion);
+                        throw new HttpRequestException(SR.net_http_retry_on_older_version, ex, RequestRetryType.RetryOnLowerHttpVersion, httpRequestError: HttpRequestError.VersionNegotiationError);
 
                     case Http3ErrorCode.RequestRejected:
                         // The server is rejecting the request without processing it, retry it on a different connection.
-                        throw new HttpRequestException(SR.net_http_request_aborted, ex, RequestRetryType.RetryOnConnectionFailure);
+                        throw new HttpRequestException(SR.net_http_request_aborted, ex, RequestRetryType.RetryOnConnectionFailure, httpRequestError: HttpRequestError.Unknown);
```

Comment:
When can these cases occur? How can we test them? Can we return something better for `HttpRequestError.Unknown`? Same for the other cases where I'm using `HttpRequestError.Unknown`.

Reply:
The way I understand it, these do not surface to the user, they mean that we should retry on a different connection (either different version, or just a brand new H/3 connection). Looking at the handling code (after some retries, they might bubble up though).
The first one gets wrapped:
https://github.com/dotnet/runtime/blob/b898a540f75d78ca55ee33288a4fd62b1ab69fa2/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs#L1165
The second won't:
https://github.com/dotnet/runtime/blob/b898a540f75d78ca55ee33288a4fd62b1ab69fa2/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs#L1142

But, it's the same thing as this:
https://github.com/dotnet/runtime/blob/b898a540f75d78ca55ee33288a4fd62b1ab69fa2/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs#L2205-L2206

**which you don't touch in this PR, so do we need to touch these at all????**

Also, you should be easily able to test it with our loopback set up:
- RequestRejected: https://github.com/dotnet/runtime/blob/d4db2d96bb0f55211c30dc69ac587b204bb418f3/src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http3.cs#L384-L384
- VersionFallback: seems like we don't have a test for it, or at least I cannot find anything.

Reply:
I handled these cases in 9afb7900d89ec6421225ea84c3b9f8185f1843b9, by using `HttpRequestError.ProtocolError` and `ProtocolException` as an inner exception, but there are some other HTTP/3 cases which are still use `Unknown`:

https://github.com/dotnet/runtime/blob/2eb3a4a8e136aeca554372d10f20d514898a56a4/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs#L277-L281

https://github.com/dotnet/runtime/blob/2eb3a4a8e136aeca554372d10f20d514898a56a4/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs#L291-L295

https://github.com/dotnet/runtime/blob/2eb3a4a8e136aeca554372d10f20d514898a56a4/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs#L302-L309

I don't know what can trigger those, and whether there is a better `HttpRequestError` we could use. /cc @wfurt